### ISSUE TITLE
 torchvision save_image produces incorrect results when saving png files

### ISSUE THREAD

**Issue by rustrust** [pytorch/pytorch#102893]
## Issue description

torchvision save_image produces incorrect results when saving png files 

in the attached images, you will see color pixels, but the input data is a rank two tensor so the images should be grayscale.

![foo1](https://github.com/pytorch/pytorch/assets/36512372/0a57e135-d099-4417-a693-b24a91a68b25)
![foo2](https://github.com/pytorch/pytorch/assets/36512372/6f78e0d4-e904-4060-a44d-9ffd7e1be97d)
![foo3](https://github.com/pytorch/pytorch/assets/36512372/6f7fa371-d161-4cf3-88d0-a0e52c0203be)
![foo4](https://github.com/pytorch/pytorch/assets/36512372/6899e235-29d6-484f-a33d-0461907613e6)

## Code example

for mnist (using mps as the device):

def test(model, device, test_loader):
    model.eval()
    test_loss = 0
    correct = 0
    with torch.no_grad():
        for data, target in test_loader:
            data, target = data.to(device), target.to(device)
            output = model(data)
            for i in range(0, 10):
               foo = data[i].reshape(28, 28).clone()
               torchvision.utils.save_image(foo, "foo" +str(i)+".png", normalize=True)

## System Info

- PyTorch or Caffe2: pytorch
- How you installed PyTorch: conda
- Build command you used: not compiled from source
- OS: macOS Ventura 13.4
- PyTorch version: 2.0.1
- Python version: 3.9.16
- CUDA/cuDNN version: n/a (using mps backend)
- GPU models and configuration: M1 cpu (mps backend)
